src/broadcast.jl

```diff
@@ -12,8 +20,14 @@ Base.similar(X::Union{pm_Vector, pm_Matrix}, ::Type{S}, dims::Dims{2}) where
 
 Base.similar(X::Union{pm_Vector, pm_Matrix}, ::Type{S}, dims::Dims{2}) where {S} = Matrix{S}(dims...)
 
+Base.similar(X::pm_SparseMatrix, ::Type{S}, dims::Dims{2}) where
+    {S <: pm_VecOrMat_eltypes} = pm_SparseMatrix{convert_to_pm_type(S)}(dims...)
+
+Base.similar(X::pm_SparseMatrix, ::Type{S}, dims::Dims{2}) where {S} = SparseMatrixCSC{S}(dims...)
```

Comment:
wouldn't that mean that broadcasting a `pm_SparseMatrix` gives us a `pm_Matrix` or a julia matrix? running the test script with this alteration yields two additional fails because we receive a dense matrix instead of a sparse one